R/preprocess.R

```diff
@@ -589,6 +589,70 @@ enw_reporting_triangle_to_long <- function(obs) {
   return(reports_long[])
 }
 
+#' Flag observed observations
+#'
+#' @description Flags observations based on the 'confirm' column.
+#' If the '.observed' column does not exist, it is created. Observations are
+#' flagged as observed (`TRUE`) if 'confirm' is not NA.
+#'
+#' @param obs A `data.frame` with at least a have `confirm` column.
+#'
+#' @return A `data.table` with an additional column '.observed' indicating
+#' observed observations.
+#'
+#' @inheritParams enw_preprocess_data
+#' @family preprocess
+#' @export
+#' @examples
+#' dt <- data.frame(id = 1:3, confirm = c(NA, 1, 2))
+#' enw_flag_observed_observations(dt)
+enw_flag_observed_observations <- function(obs, copy = TRUE) {
```

Comment:
Also, admittedly, `flag_observed_observations` could be a bit confusing? How about just `flag_observed` or `flag_reported_observations`?

Reply:
Yeah you are right it isn't a good name... Maybe just `flag_observed` is enough?